build.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+echo "--- Setup"
+
+git submodule update --init
+
+################################################################################
+# Cabal config
+
+cabal_args=()
+if [ -n "${CABAL_STORE_DIR:-}" ]; then
+  cabal_args+=("--store-dir=$CABAL_STORE_DIR")
+fi
+
+################################################################################
+# Caching
+
+restore_cache() {
+  test -n "$cache_args" || exit 0
+  echo "--- Restoring from cache..."
+  cabal-cache sync-from-archive "${cache_args[@]}" || true
```

Comment:
What's the reason for the `|| true`.  Is this expected to fail under some circumstances and ignored?

Reply:
Yeah I wouldn't expect it to fail, but don't want any cache sync problems to stop a build.